cpp/src/replace/replace.cu

```diff
@@ -87,6 +90,108 @@ __device__ auto get_new_value(cudf::size_type idx,
   return thrust::make_pair(new_value, output_is_valid);
 }
 
+__device__ int get_new_string_value(cudf::size_type idx,
```

Comment:
`get_new_value` should be updated and then there is no need for two different functions. It should look like:
```c++
template <typename Element, bool replacement_has_nulls>
__device__ auto get_new_value(size_type i, column_device_view input, cudf::column_device_view values_to_replace, column_device_view replacement_values){

   auto found = thrust::find(thrust::seq, values_to_replace.begin<Element>(), values_to_replace.end<Element>(), input.element<Element>());
   
   Element new_value;
   bool output_is_valid{true};
   if(found != values_to_replace.end<Element>()){
      auto d = thrust::distance(values_to_replace.begin<Element>(), found);
      new_value = replacement_values.element<Element>(d);
      if(replacement_has_nulls){
         output_is_valid = replacement_values.is_valid(d);
      }
   } else {
      new_value = input.element<Element>(i);
   }
   return thrust::make_pair(new_value, output_is_valid);
}
```

Otherwise separate function is just extra baggage that needs to be maintained.

Reply:
The reason for having two functions is that the one for strings returns the index of the found value, rather than the value itself. This prevents the second pass kernel from needing to do a search again.